crates/rattler/src/solver/libsolv/repo.rs

```diff
@@ -173,21 +173,22 @@ impl RepoRef {
                 )
             };
 
-            // Location (filename (fn) and subdir)
-            let filename = record
-                .filename
-                .as_deref()
-                .unwrap_or_else(|| filename.as_ref());
-            let subdir = record.subdir.as_str();
-            unsafe {
-                ffi::repodata_set_location(
-                    data,
-                    solvable_id.into(),
-                    0,
-                    CString::new(subdir)?.as_ptr(),
-                    CString::new(filename)?.as_ptr(),
-                );
-            }
+            // TODO: The filename is not present in a PackageRecord
```

Comment:
Although the filename is the key to the record in the `repodata.json`, right? Maybe we could add that in? On the other hand, that information will be in the repodata record, so that is fine. I just wonder if repodata record is more expensive to construct

Reply:
Yeah, I wanted to keep the `PackageRecord` "pure" in that it represents exactly what is stored in the `repodata.json`. You can quickly convert the `RepoDataRecord` from this and its original channel because you can move the `PackageRecord` directly in it. However, storing all the URLs and channels etc does have some memory overhead. Maybe we can add some form of interning in the future?

Reply:
Also for the solver, it now ingests both the `RepoData` with the `PackageRecords` and `Channel` where it comes from. I think we can replace that with simply passing in `RepoDataRecord` because that then contains all the information.